sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/HiveOrcSourceSuite.scala

```diff
@@ -153,4 +153,18 @@ class HiveOrcSourceSuite extends OrcSuite with TestHiveSingleton {
       }
     }
   }
+
+  test("SPARK-33755: Allow creating orc table when row format separator is defined") {
+    withTable("row_format_orc") {
+      sql(
+        s"""CREATE TABLE row_format_orc(
+           |  intField INT,
+           |  stringField STRING
+           |)
+           |ROW FORMAT DELIMITED FIELDS TERMINATED BY '002'
```

Comment:
How does the ORC table wotk with the delimiter?

Reply:
Although this unnecessary for ORC table, we can support an option that user can choose to ignore it or not.

Reply:
It's better to be explicit for what Spark supports. It's a bit odd that we add no-op syntax.

Reply:
We have many tasks which are running on hive and slowly, so we want to migrate it from hive to spark. I think other companies want to this too. If we support ignore this exception by set spark.sql.orc.skipRowFormatDelimitedError=true , we can migrate hive task and user do not need to modify his sql script.

Reply:
Does Hive work with this delimiter specified with ORC?

Reply:
Hive doesn't work with it, but hive doesn't throw exception too.

Reply:
Let's don't fix it then. It's odd that Spark works with a syntax that's no-op, and that does not also work in Hive.

Reply:
thx very much, i'll close it.